### ISSUE TITLE
🐛 Postgres source returns null for Infinity, -Infinity or Nan

### ISSUE THREAD

**Issue by tuliren** [airbytehq/airbyte#8902]
We can correctly parse these special values by:

1. Add the following method to `PostgresSourceOperations`:

```java
protected void putBigDecimal(final ObjectNode node, final String columnName, final ResultSet resultSet, final int index) throws SQLException {
  final BigDecimal bigDecimal = DataTypeUtils.returnNullIfInvalid(() -> resultSet.getBigDecimal(index));
  if (bigDecimal != null) {
    node.put(columnName, bigDecimal);
    return;
  }

  // ResultSet#getBigDecimal cannot handle Infinity, -Infinity, or NaN, and will throw exception,
  // which becomes null. So we need to check these special values as string.
  final String value = resultSet.getString(index);
  if (value.equalsIgnoreCase("infinity")) {
    node.put(columnName, Double.POSITIVE_INFINITY);
  } else if (value.equalsIgnoreCase("-infinity")) {
    node.put(columnName, Double.NEGATIVE_INFINITY);
  } else if (value.equalsIgnoreCase("nan")) {
    node.put(columnName, Double.NaN);
  } else {
    node.put(columnName, (BigDecimal) null);
  }
}
```

2. Revert https://github.com/airbytehq/airbyte/pull/8726/commits/99267234bcfd49ced989f4945beb2e06ef7ecc19

However, this change is reverted because Json numbers do not support these special values, and if we set those fields to the special values, those fields become string fields. It is unclear how the destination / dbt will handle them. See discussion [here](https://github.com/airbytehq/airbyte/pull/8726#discussion_r769479251).

**Comment by tuliren** [comment]
Special number value handling in normalization:
https://github.com/airbytehq/airbyte/pull/8726#discussion_r772596643

**Comment by edgao** [comment]
@grishick I just chatted with @alexandertsukanov ; summarizing:

* IMO this issue is going to be challenging to solve with our current type system, and should be postponed until we have something more robust
* The proposal @alexandertsukanov came up with is to modify the protocol. `AirbyteRecordMessage` would have the ability to force destination connectors to interpret a value in a different way, e.g:
   ```json
   {
     "data": {
       "foo_column": 0.0,
       "foo_column_identifier": "+Infinity"
     }
   }
   ```
   This would require changes in normalization, along with many destination connectors, to handle the "double-entry" style, and represents a philosophical change (that the `data` entry in an AirbyteRecordMessage is mostly-opaque)
* The other option would be to dump all numeric values into strings, which completely sidesteps the problem that `{"foo": infinity}` is invalid json. This is effectively approach 3 on the issue the team talked about on Tuesday (https://github.com/airbytehq/airbyte/issues/12606#issuecomment-1129155345); we probably don't want to do it right now.
    * There's no equivalent to approaches 1 or 2, because json literally cannot represent these values natively

**Comment by grishick** [comment]
I am going to modify this [Epic](https://github.com/airbytehq/airbyte/issues/9441) to include implementing approach 3 discussed in [this issue](https://github.com/airbytehq/airbyte/issues/12606#issuecomment-1129131814). As we discussed during backlog grooming, that approach will require sweeping changes across all destinations and normalization, so we'll table it for now until we have cleared the path for releasing Postgres Source to GA. I think this is still a fairly high priority problem, so we'll pick it back up soon.

**Comment by grishick** [comment]
I moved this to On Hold until we are ready for [approach 3](https://github.com/airbytehq/airbyte/issues/12606#issuecomment-1129131814)

**Comment by grishick** [comment]
We cannot fix this until we have a better solution for sources to convey specific types to destinations and normalization

**Comment by bleonard** [comment]
@akashkulk @edgao I'm pretty sure this is on track to be completed as part of the data types work. That right?

**Comment by akashkulk** [comment]
That's correct - once all of the messages from the source have been upgraded, this will be a quick change.

**Comment by bleonard** [comment]
Duplicate of https://github.com/airbytehq/airbyte/issues/19952

**Comment by bleonard** [comment]
Saved me! Thanks, @edgao
